poly-org.el

```diff
@@ -70,8 +81,14 @@ Used in :switch-buffer-functions slot."
   :fallback-mode 'host
   :head-mode 'host
   :tail-mode 'host
-  :head-matcher "^[ \t]*#\\+begin_src .*\n"
-  :tail-matcher "^[ \t]*#\\+end_src"
+  ;; the following is incorrect:
+  ;; tail-matcher actually depends on head-matcher
```

Comment:
Tail matcher can depend on the front matcher because it's called "with the point at the end of the matched head" So you can easily infer the type of the block. 

One solution would be to replace the tail matcher with a function which would infer the type of the head. Another solution would be to have a head-matcher a function as well and store the head type globally or as a text property. I think the former solution is cleaner, even though it requires an extra backward regexp lookup.

Reply:
The proper way would be to share the scope of pattern variables between two expressions. I'm not ready to dig into this at the moment. I'm not even sure this makes sense in terms of regexps.

Reply:
No, that's too complex and can be already done satisfactory.